vm/proxyapp/proxyrpc/proxyrpc.go

```diff
@@ -0,0 +1,93 @@
+// Copyright 2022 syzkaller project authors. All rights reserved.
+// Use of this source code is governed by Apache 2 LICENSE that can be found in the LICENSE file.
+
+package proxyrpc
+
+import (
+	"time"
+)
+
+type ConstructPoolParams struct {
+	Debug bool
+	Param string
+}
+
+type ConstructPoolResult struct {
+	Count int // signal the created pool size
+}
+
+type CreateInstanceParams struct {
+	Workdir string
+	Index   int
+}
+
+type CreateInstanceResult struct {
+	ID string // allocated instance id
+}
+
+type MethodCopyParams struct {
+	ID      string
+	HostSrc string
+}
+
+type MethodCopyResult struct {
+	VMFileName string
+}
+
+type MethodForwardParams struct {
+	ID   string
+	Host string // currently is not used. Proxyapp use config param here.
+	Port int
+}
+
+type MethodForwardResult struct {
+	ManagerAddress string
+}
+
+type MethodRunStartParams struct {
+	ID      string
+	Timeout time.Duration
+	Command string
+}
+
+type MethodRunStartReply struct {
+	RunID string
+}
+
+type MethodRunStopParams struct {
+	ID    string
+	RunID string
+}
+
+type MethodRunStopReply struct {
+}
+
+type MethodRunReadProgressParams struct {
+	ID    string
+	RunID string
+}
+
+type MethodRunReadProgressReply struct {
+	FuzzerStdoutChunk string
+	FuzzerStderrChunk string
+	ConsoleOutChunk   string
```

Comment:
Do these fields have to be strings? Maybe it would be easier to use `[]byte` arrays?

Reply:
We want plugins to be implemented in different languages.
Lets use as less base types in proxyrpc.go as possible to simplify the plugins protocol implementation.

Reply:
Is []byte encoded differently and is harder to handle?

Reply:
[Quick check](https://go.dev/play/p/zCzHON-L7WB) for json.Marshal().
{"Bytes":"Ynl0ZXM=","Str":"string"}
[]byte requires +base64.

Internally [jsonrpc use json.Unmarshal()](https://cs.opensource.google/go/go/+/refs/tags/go1.19.1:src/net/rpc/jsonrpc/client.go;l=104)